app/controllers/floors_controller.rb

```diff
@@ -54,77 +49,37 @@ def floor_params
 
   private
 
-  def fetch_persistent_locations
-    Location.where(:persistent => true)
-  end
-
   def process_search(search_params)
-    @search_results = []
```

Comment:
This seems like a good refactor because it puts all the search logic in one place. Instead of making a decision on whether or not we should run this method, it always runs and then quickly returns an empty array if no search happened. Otherwise, it'll run the search on all the object types.